schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/spi/ActivityCreator.java

```diff
@@ -59,14 +64,37 @@ private Activity createActivity(Activity from, ActivityType activityType, Repeat
         return newActivity;
     }
 
+    /**
+     *
+     * @param fromDateTime: DateTime to increment
+     * @param days: Number of days to increment
+     * @return fromDateTime.plusDays(days). If increment moves over daylight saving interval will adjust hour value appropriately
+     */
+    private DateTime increment(DateTime fromDateTime, int days) {
+        DateTime toDateTime = fromDateTime.plusDays(days);
+
+        // If the offset changes we must have moved to the next interval
+        long fromHourOffset = TimeUnit.MILLISECONDS.toHours(ZurichZone.getOffset(fromDateTime.getMillis()));
+        long toHourOffset = TimeUnit.MILLISECONDS.toHours(ZurichZone.getOffset(toDateTime.getMillis()));
+
+        if (fromHourOffset != toHourOffset) {
+            int offset = (int) (fromHourOffset - toHourOffset);
+            toDateTime = toDateTime.plusHours(offset);
+        }
+
+        return toDateTime;
+    }
```

Comment:
I don't know joda time that well, but I would expect that it should be possible to do something like:
~~~java
DateTime toDateTime = fromDateTime.plusDays(days);
if( fromDateTime.getHour() != toDateTime.getHour() ) {
  toDateTime.setHour(fromDateTime.getHour());
}
~~~